File: Mediator/Golang/CreateCommand.go
package main

import "fmt"

type CreateCommand struct {
	mediator 	*LockingSystem
	file 		string
}

func (command *CreateCommand) execute() {
	if !command.mediator.checkStatus() {
		command.mediator.lock(command)
		command.describe()
		fmt.Println("Command executed")
	} else {
		fmt.Println("Command not executed")
	}
}

func (command *CreateCommand) finish() {
	command.mediator.unlock()
}

func (command *CreateCommand) describe() {
	fmt.Println("Creating file ", command.file)
}

File: Mediator/PHP/CreateCommand.php
<?php

require_once "Command.php";
require_once "LockingSystem.php";

class CreateCommand extends Command {
    public function __construct(LockingSystem $mediator, string $file)
    {
        parent::__construct($mediator, $file);
    }

    public function describe(){
        echo "Creating file " . $this->file . "\n";
    }
}